src/Void.Core/Messaging.fs

```diff
@@ -0,0 +1,216 @@
+﻿namespace Void.Core
+
+module Messaging = 
+    type private Channel =
+        abstract member publish : Message -> Message seq
+        (* F# Why you no have type classes like Haskell!?!?!
+         * Now I will do ugly things, with long names! *)
+        abstract member getBoxedSubscribeActionIfTypeIs<'TMsg> : unit -> obj option
+        
+    type private Channel<'TIn when 'TIn :> Message>
+        (
+            handlers : Handle<'TIn> list
+        ) =
+        let mutable _handlers = handlers
+
+        member private x.safetyWrap handle message =
+            try
+                handle message
+            with ex ->
+                printf "Error while handling %A: %A" message ex
+                noMessage
+
+        member x.addHandler handler =
+            _handlers <- x.safetyWrap handler :: _handlers
+
+        interface Channel with
+            member x.publish (message : Message) =
+                match message with
+                    | :? 'TIn as msg ->
+                        Seq.map (fun handle -> handle msg) _handlers
+                        |> Seq.filter (fun msg -> msg <> noMessage)
+                    | _ -> Seq.empty
+
+            member x.getBoxedSubscribeActionIfTypeIs<'TMsg>() =
+                if typeof<'TIn> = typeof<'TMsg>
+                then Some <| box x.addHandler
+                else None
+
+    type private RequestChannel =
+        (* F# Why you no have type classes like Haskell!?!?!
+         * Now I will do ugly things, with long names! *)
+        abstract member getBoxedRequestFunctionIfResponseTypeIs<'TMsg> : unit -> obj option
+        abstract member getBoxedSubscribeActionIfResponseTypeIs<'TMsg> : unit -> obj option

```

Comment:
Yeah, me too. If you have a good distinguishing rename suggestion, I'd be quite open to it.